docs/azureai/azureai-search-document-integration.md

```diff
@@ -1,22 +1,175 @@
+
+There are many more configuration options available to customize the Azure AI Search resource. For more information, see [`Azure.Provisioning` customization](../azure/integrations-overview.md#azureprovisioning-customization).
+
+### Connect to an existing Azure AI Search service
+
+You might have an existing Azure AI Search service that you want to connect to. You can chain a call to annotate that your <xref:Aspire.Hosting.Azure.AzureSearchResource> is an existing resource:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var existingSearchName = builder.AddParameter("existingSearchName");
+var existingSearchResourceGroup = builder.AddParameter("existingSearchResourceGroup");
+
+var search = builder.AddAzureSearch("search")
+                    .AsExisting(existingSearchName, existingSearchResourceGroup);
+
+builder.AddProject<Projects.ExampleProject>()
+       .WithReference(search);
+
+// After adding all resources, run the app...
+```
+
+For more information on treating Azure AI Search resources as existing resources, see [Use existing Azure resources](../azure/integrations-overview.md#use-existing-azure-resources).
+
+Alternatively, instead of representing an Azure AI Search resource, you can add a connection string to the app host. Which is a weakly-typed approach that's based solely on a `string` value. To add a connection to an existing Azure AI Search service, call the <xref:Aspire.Hosting.ParameterResourceBuilderExtensions.AddConnectionString%2A> method:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var search = builder.ExecutionContext.IsPublishMode
+    ? builder.AddAzureSearch("search")
+    : builder.AddConnectionString("search");
+
+builder.AddProject<Projects.ExampleProject>()
+       .WithReference(search);
+
+// After adding all resources, run the app...
+```
+
+[!INCLUDE [connection-strings-alert](../includes/connection-strings-alert.md)]
+
+The connection string is configured in the app host's configuration, typically under User Secrets, under the `ConnectionStrings` section:
+
+```json
+{
+  "ConnectionStrings": {
+    "search": "https://{account_name}.search.azure.com/"
+  }
+}
+```
+
+For more information, see [Add existing Azure resources with connection strings](../azure/integrations-overview.md#add-existing-azure-resources-with-connection-strings).
+
+### Conditional configuration based on execution mode
+
+If you want to use Azure AI Search when publishing to Azure but use a connection string during local development, you can use the `ExecutionContext.IsPublishMode` property:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var search = builder.ExecutionContext.IsPublishMode
+    ? builder.AddAzureSearch("search")
+    : builder.AddConnectionString("search");
+
+var myService = builder.AddProject<Projects.MyService>()
+                       .WithReference(search);
+
+// After adding all resources, run the app...
```

Comment:
Isn't this a duplicate of the above "Alternatively, instead of representing an Azure AI Search resource, you can add a connection string to the app host." part?

Reply:
Yes, good catch. Removing that.